addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4150,17 +4150,132 @@ QUnit.test('messages marked as read move to "History" mailbox', async function (
                 this.env.messaging.history.localId
             }"]
         `).classList.contains('o-active'),
-        "History mailbox should be active"
+        "history mailbox should be active"
     );
     assert.strictEqual(
         document.querySelectorAll(`.o_Discuss_thread .o_MessageList_empty`).length,
         0,
-        "History mailbox should not be empty after mark as read"
+        "history mailbox should not be empty after mark as read"
     );
     assert.strictEqual(
         document.querySelectorAll(`.o_Discuss_thread .o_MessageList_message`).length,
         2,
-        "History mailbox should have 2 messages"
+        "history mailbox should have 2 messages"
+    );
+});
+
+QUnit.test('single message marked as read should move to "History" mailbox and not others', async function (assert) {
```

Comment:
mark a single message as read should only move this message to "History" mailbox